flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/sink/Semantic.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.connectors.kafka.sink;
+
+/**
+ * Semantics that can be chosen.
+ * <li>{@link #EXACTLY_ONCE}
+ * <li>{@link #AT_LEAST_ONCE}
+ * <li>{@link #NONE}
+ */
+public enum Semantic {
```

Comment:
I know that this is the old name, but isn't this more ProcessingGuarantees? Should we factor that out to `flink-core`?

Reply:
Why do you opt for `flink-core` and not something like `connector-base`? In general, I like the idea to move it somewhere reusable across multiple connectors.

Reply:
`connector-base` are building blocks, `flink-core` is the core API. I'm not 100%, so maybe it's a good idea to start with `connector-base` and uplevel when needed.
Btw I also like to provide base sink/source builder where we could already implement the property.

Reply:
I moved it to `connector base` for now and renamed it.